ipamutils/utils.go

```diff
@@ -37,18 +37,37 @@ type NetworkToSplit struct {
 // InitNetworks initializes the broad network pool and the granular network pool
 func InitNetworks(defaultAddressPool []*NetworkToSplit) {
 	initNetworksOnce.Do(func() {
-		// error ingnored should never fail
-		PredefinedGranularNetworks, _ = splitNetworks(defaultGranularNetwork)
+		mutex.Lock()
```

Comment:
looks like the lock here can be removed and you can directly call the method `ConfigGlobalScopeDefaultNetworks`

Reply:
Idea is ConfigGlobalScopeDefaultNetworks is invoked only from swarmkit and init network will be invoked only during Daemon Init.   Later if we plan to support  swarmkit update , we dont can modify ConfigGlobalScopeDefaultNetworks without worrying about InitNetworks calling this API.  would like to keep it that way .

Reply:
If you want to keep it this way I think you should add a comment explaining why.  Otherwise someone may come along later and refactor `InitNetworks()` to call `ConfigGlobalScopeDefaultNetworks()` as the logic is (more or less) duplicated.